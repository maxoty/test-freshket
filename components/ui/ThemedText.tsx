import { Text, type TextProps, StyleSheet } from "react-native";

import { useThemeColor } from "@/hooks/useThemeColor";

export type ThemedTextProps = TextProps & {
  lightColor?: string;
  darkColor?: string;
  type?:
    | "small"
    | "smallMedium"
    | "default"
    | "title"
    | "defaultMedium"
    | "subtitle"
    | "link"
    | "button"
    | "headline";
  variant?: "default" | "primary" | "primarySoft" | "secondary" | "error";
};

export function ThemedText({
  style,
  lightColor,
  darkColor,
  type = "default",
  variant = "default",
  ...rest
}: ThemedTextProps) {
  const colorName = (variant: string) => {
    switch (variant) {
      case "primary":
        return "textPrimary";
      case "primarySoft":
        return "textPrimarySoft";
      case "secondary":
        return "textSecondary";
      case "error":
        return "error";
      default:
        return "text";
    }
  };

  const color = useThemeColor(
    { light: lightColor, dark: darkColor },
    colorName(variant)
  );

  return (
    <Text
      style={[
        { color },
        type === "small" ? styles.small : undefined,
        type === "smallMedium" ? styles.smallMedium : undefined,
        type === "default" ? styles.default : undefined,
        type === "title" ? styles.title : undefined,
        type === "defaultMedium" ? styles.defaultMedium : undefined,
        type === "subtitle" ? styles.subtitle : undefined,
        type === "link" ? styles.link : undefined,
        type === "button" ? styles.button : undefined,
        type === "headline" ? styles.headline : undefined,
        style,
      ]}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  small: {
    fontFamily: "Roboto",
    fontSize: 14,
    lineHeight: 20,
    fontWeight: "400",
    letterSpacing: 0.1,
  },
  smallMedium: {
    fontFamily: "RobotoMedium",
    fontSize: 14,
    lineHeight: 20,
    fontWeight: "500",
    letterSpacing: 0.1,
  },
  default: {
    fontFamily: "Roboto",
    fontSize: 16,
    lineHeight: 24,
    fontWeight: "400",
    letterSpacing: 0.15,
  },
  defaultMedium: {
    fontFamily: "RobotoMedium",
    fontSize: 16,
    lineHeight: 24,
    fontWeight: "500",
    letterSpacing: 0.15,
  },
  title: {
    fontFamily: "Roboto",
    fontSize: 22,
    lineHeight: 28,
    fontWeight: "400",
    letterSpacing: 0,
  },
  subtitle: {
    fontFamily: "Roboto",
    fontSize: 20,
    fontWeight: "400",
  },
  link: {
    fontFamily: "Roboto",
    lineHeight: 30,
    fontSize: 16,
    fontWeight: "400",
  },
  button: {
    fontFamily: "RobotoMedium",
    fontSize: 14,
    lineHeight: 20,
    fontWeight: "500",
    letterSpacing: 0.1,
  },
  headline: {
    fontFamily: "Roboto",
    fontSize: 32,
    lineHeight: 40,
    fontWeight: "400",
  },
});
