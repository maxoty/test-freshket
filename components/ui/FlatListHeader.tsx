import { StyleSheet } from "react-native";
import { ThemedText } from "./ThemedText";
import { ThemedView } from "./ThemedView";

export type FlatListHeaderProps = {
  title: string;
};

export function FlatListHeader({ title }: FlatListHeaderProps) {
  return (
    <ThemedView style={styles.container}>
      <ThemedText type="title">{title}</ThemedText>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 16,
    paddingVertical: 8,
    height: 64,
    justifyContent: "center",
  },
});
