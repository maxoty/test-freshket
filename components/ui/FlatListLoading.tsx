import { ActivityIndicator, StyleSheet, View } from "react-native";
import { ThemedText } from "./ThemedText";
import { Colors } from "@/constants/Colors";

export type FlatListLoadingProps = {
  loading: boolean;
};

export function FlatListLoading({ loading }: FlatListLoadingProps) {
  if (!loading) return;

  return (
    <View style={styles.container}>
      <ActivityIndicator size={26} color={Colors.light.primary} />
      <ThemedText type="smallMedium">Loading..</ThemedText>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 90,
    paddingVertical: 20,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    gap: 8,
  },
});
