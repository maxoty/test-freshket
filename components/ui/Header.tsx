import { StyleSheet, TouchableOpacity } from "react-native";
import { ThemedText } from "./ThemedText";
import { ThemedView } from "./ThemedView";
import { IconSymbol } from "./IconSymbol";
import { Colors } from "@/constants/Colors";

export type HeaderProps = {
  title: string;
  onPressBack?: () => void;
};

export function Header({ title, onPressBack }: HeaderProps) {
  return (
    <ThemedView style={styles.container}>
      <TouchableOpacity style={styles.button} onPress={() => onPressBack?.()}>
        <IconSymbol name="arrow.left" color={Colors.light.text} />
      </TouchableOpacity>
      <ThemedText type="title">{title}</ThemedText>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 4,
    paddingVertical: 6,
    height: 64,
    flexDirection: "row",
    alignItems: "center",
  },
  button: {
    width: 48,
    height: 48,
    alignItems: "center",
    justifyContent: "center",
  },
});
