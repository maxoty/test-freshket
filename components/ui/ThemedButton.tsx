import { StyleSheet, TouchableOpacity } from "react-native";

import { useThemeColor } from "@/hooks/useThemeColor";
import { ThemedText } from "./ThemedText";
import { PropsWithChildren } from "react";
import { TouchableOpacityProps } from "react-native-gesture-handler";
import { Colors } from "@/constants/Colors";

export type ThemedButtonProps = PropsWithChildren<TouchableOpacityProps> & {
  lightColor?: string;
  darkColor?: string;
  type?: "default" | "icon";
  variant?: "primary" | "secondary";
};

export function ThemedButton({
  style,
  lightColor,
  darkColor,
  type = "default",
  variant = "primary",
  children,
  ...rest
}: ThemedButtonProps) {
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "primary"
  );

  return (
    <TouchableOpacity
      activeOpacity={0.8}
      style={[
        { backgroundColor },
        type === "default" ? styles.default : undefined,
        type === "icon" ? styles.icon : undefined,
        style,
      ]}
      {...rest}
    >
      <ThemedText
        type="button"
        style={[
          {
            color: variant === "primary" ? Colors.dark.text : Colors.light.text,
            textAlign: "center",
          },
        ]}
      >
        {children}
      </ThemedText>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  default: {
    height: 40,
    borderRadius: 100,
    paddingHorizontal: 24,
    paddingVertical: 10,
  },
  icon: {
    width: 36,
    height: 36,
    borderRadius: 100,
    alignItems: "center",
    justifyContent: "center",
  },
});
