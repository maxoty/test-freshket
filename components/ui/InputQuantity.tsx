import { StyleSheet, View } from "react-native";
import { ThemedText } from "./ThemedText";
import { ThemedButton } from "./ThemedButton";
import { IconSymbol } from "./IconSymbol";
import { Colors } from "@/constants/Colors";

export type InputQuantityProps = {
  quantity: number;
  onChange: (quantity: number) => void;
};

export function InputQuantity({ quantity, onChange }: InputQuantityProps) {
  return (
    <View style={styles.container}>
      <ThemedButton type="icon" onPress={() => onChange(quantity - 1)}>
        <IconSymbol size={12} name="minus" color={Colors.dark.text} />
      </ThemedButton>
      <ThemedText type="defaultMedium" style={styles.quantity}>
        {quantity}
      </ThemedText>
      <ThemedButton type="icon" onPress={() => onChange(quantity + 1)}>
        <IconSymbol size={12} name="plus" color={Colors.dark.text} />
      </ThemedButton>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    gap: 4,
    width: 120,
  },
  quantity: {
    width: 28,
    textAlign: "center",
  },
});
