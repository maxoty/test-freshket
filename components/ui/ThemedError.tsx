import { StyleSheet } from "react-native";
import { ThemedText } from "./ThemedText";
import { ThemedView } from "./ThemedView";
import { IconSymbol } from "./IconSymbol";
import { Colors } from "@/constants/Colors";
import { ThemedButton } from "./ThemedButton";

export type ThemedErrorProps = {
  title: string;
  onClick?: () => void;
};

export function ThemedError({ title, onClick }: ThemedErrorProps) {
  return (
    <ThemedView style={styles.container}>
      <IconSymbol name="x.circle" color={Colors.light.error} size={54} />
      <ThemedText type="title">{title}</ThemedText>
      {onClick && (
        <ThemedButton type="default" onPress={onClick}>
          Refresh
        </ThemedButton>
      )}
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
    paddingVertical: 24,
    gap: 10,
    alignItems: "center",
    justifyContent: "center",
  },
});
