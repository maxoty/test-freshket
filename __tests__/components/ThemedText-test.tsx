import { ThemedText } from '@/components/ui/ThemedText';
import * as React from 'react';
import renderer from 'react-test-renderer';


it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedText>Snapshot test!</ThemedText>).toJSON();

  expect(tree).toMatchSnapshot();
});
