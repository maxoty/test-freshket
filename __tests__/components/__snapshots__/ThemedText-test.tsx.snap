// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#1D1B20",
      },
      undefined,
      undefined,
      {
        "fontFamily": "Roboto",
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0.15,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
